src/py42/_internal/archive_access.py

```diff
@@ -77,7 +77,10 @@ def _get_file_via_walking_tree(self, file_path):
             if root[u"path"].lower() == path_root.lower():
                 return self._walk_tree(root, path_parts[1:])
 
-        raise Py42ArchiveFileNotFoundError(self._device_guid, file_path)
+        message = u"File not found in archive for device {0} at path {1}".format(
```

Comment:
This is code is shared with what is on line 96 and can be replaced with a helper function that creates the exception

Reply:
Or, another custom exception would also work

Reply:
I would put `Py42ArchiveFileNotFoundError` back the way it was before and either make a new exception type or just use `Py42Error`.

Reply:
Done.